### PR TITLE
Simplify bangify methods in guild cache.

### PR DESCRIPTION
**Pull request by jchristgit**
Guild caches provide banged methods such as `get!/1`, `select!/2` and
friends, which were implemented directly in the backing guild cache. It
makes more sense to define these once in the dispatcher to prevent
having to duplicate logic across cache backends. This is already being
performed in the pluggable user cache.

### CODE REVIEW

**Comment by jchristgit** [comment]
Hi @jb3, any chance you have time to review this this weekend? Thank you!

**Review by Th3-M4jor — APPROVED** [review]
Looks good to me, though I do have one or two minor nitpicks on the helper functions you added.

Could you change the return types in the typespecs on them to be consistent with their parentheses? Same with them all being one-liners or not?

File: lib/nostrum/cache/guild_cache.ex [hunk 1]
```diff
@@ -344,4 +313,36 @@ defmodule Nostrum.Cache.GuildCache do
...
+  @doc """
+  Same as `get_by/1`, but raises `Nostrum.Error.CacheError` in case of failure.
+  """
+  @spec get_by!(clauses) :: Guild.t() | no_return()
+  def get_by!(clauses), do: clauses |> get_by |> Util.bangify_find(clauses, __MODULE__)
+
+  @doc """
+  Same as `select/2`, but raises `Nostrum.Error.CacheError` in case of failure.
+  """
+  @spec select!(Guild.id(), selector) :: any | no_return
```
Comment (jchristgit) [inline lib/nostrum/cache/guild_cache.ex:334]:
```suggestion
  @spec select!(Guild.id(), selector) :: any() | no_return()
```

**Comment by jchristgit** [comment]
> Looks good to me, though I do have one or two minor nitpicks on the helper functions you added.
> 
> Could you change the return types in the typespecs on them to be consistent with their parentheses? Same with them all being one-liners or not?

Done. Thank you for the review!

**Comment by jchristgit** [comment]
Thanks!

**Pull request merged**